soh/src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -6301,11 +6301,17 @@ s32 func_8083BBA0(Player* this, PlayState* play) {
 }
 
 void Player_SetupRoll(Player* this, PlayState* play) {
-    Player_SetupAction(play, this, Player_Action_Roll, 0);
-    LinkAnimation_PlayOnceSetSpeed(play, &this->skelAnime,
-                                   GET_PLAYER_ANIM(PLAYER_ANIMGROUP_landing_roll, this->modelAnimType),
-                                   1.25f * sWaterSpeedFactor);
-    gSaveContext.ship.stats.count[COUNT_ROLLS]++;
+    if (!GameInteractor_Should(VB_ROLL, true)) {
+        return false;
+    }
+
+    else {
+        Player_SetupAction(play, this, Player_Action_Roll, 0);
+        LinkAnimation_PlayOnceSetSpeed(play, &this->skelAnime,
+                                       GET_PLAYER_ANIM(PLAYER_ANIMGROUP_landing_roll, this->modelAnimType),
+                                       1.25f * sWaterSpeedFactor);
+        gSaveContext.ship.stats.count[COUNT_ROLLS]++;
+    }
```

Comment:
```suggestion
    if (GameInteractor_Should(VB_ROLL, true)) {
        Player_SetupAction(play, this, Player_Action_Roll, 0);
        LinkAnimation_PlayOnceSetSpeed(play, &this->skelAnime,
                                       GET_PLAYER_ANIM(PLAYER_ANIMGROUP_landing_roll, this->modelAnimType),
                                       1.25f * sWaterSpeedFactor);
        gSaveContext.ship.stats.count[COUNT_ROLLS]++;
    }
```
function is `void`, tho why here instead of `Player_TryRoll` which it seemed you were originally going for with this return?

Reply:
thanks